_security/access-control/api.md

```diff
@@ -1297,6 +1297,91 @@ PATCH _plugins/_security/api/securityconfig
 }
 ```
 
+### Configuration upgrade check
+Introduced 2.13
+{: .label .label-purple }
+
+Checks the configuration bundled with the Security plugin against the existing security configuration, returns if an upgrade can be performed and what resources would be updated.
+
+Between new OpenSearch version new defaults to the security configuration are added.  This endpoint helps cluster operators check if the cluster is missing defaults or has stale definitions of defaults. 
+{: .note}
+
+#### Request
+
+```json
+GET _plugins/_security/api/_upgrade_check
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+```json
+{
+  "status" : "OK",
+  "upgradeAvailable" : true,
+  "upgradeActions" : {
+    "roles" : {
+      "add" : [ "flow_framework_full_access" ]
+    }
+  }
+}
+```
+
+#### Response fields
+
+| Field    | Data type  | Description                   |
+|:---------|:-----------|:------------------------------|
+| `upgradeAvailable`   | Boolean     | True when an upgrade can be performed.  |
+| `upgradeActions`  | Object     | The security objects that would be modified by an upgrade.  |
+
+### Configuration upgrade
+Introduced 2.13
+{: .label .label-purple }
+
+Adds and updates resources on the existing security configuration from the configuration bundled with the Security plugin.
+
+These bundled configuration files can be found in the `<OPENSEARCH_HOME>/security/config` directory.  Default configuration files are updated on OpenSearch upgrade, whereas the cluster configuration is only updated by the cluster operators.  This endpoint helps cluster operators upgrade missing defaults and stale defaults definitions. 
+
+Between new OpenSearch version new defaults to the security configuration are added. This endpoint helps cluster operators check if the cluster is missing default roles or has stale definitions of default roles. 
+{: .note}
+
+#### Request
+
+```json
+POST _plugins/_security/api/_upgrade_perform
+{
+  "configs" : [ "roles" ]
+}
+```
+{% include copy-curl.html %}
+
+#### Request fields
+
+| Field           | Data type  | Description                                                                                                       | Required |
+|:----------------|:-----------|:------------------------------------------------------------------------------------------------------------------|:---------|
+| configs              | Array     | Specifies the configurations to be upgraded. This field can include any combination of the following configurations `actiongroups`,`allowlist`, `audit`, `internalusers`, `nodesdn`, `roles`, `rolesmappings`, `tenants`.<br>  Default is all supported configurations.  | No      |
```

Comment:
`Default is all supported configurations` to `If none specified, all supported configurations will be included`. Thoughts?

Reply:
Thanks for the suggestion, I think its fine as is.